src/hydration/hydration.ts

```diff
@@ -0,0 +1,127 @@
+import { DEFAULT_STALE_TIME, DEFAULT_CACHE_TIME } from '../core/config'
+
+import type { Query, QueryCache, QueryKey, QueryConfig } from 'react-query'
+
+export interface DehydratedQueryConfig<TResult> {
```

Comment:
Maybe also include the query key? It might not be possible to convert a hash back into a query key and each query can have it's own `queryKeySerializerFn`.

Reply:
This is a very good point and would let us remove the `queryHash` from the payload instead, as well as get rid of `queryKeyParserFn` which I've never liked. It requires changing the format around a little bit, I'll get on it!

Reply:
Fixed.